frontend/src/metabase/components/Schedule/GroupControlsTogether.tsx

```diff
@@ -41,11 +43,31 @@ export const GroupControlsTogether = ({
 
   const compactChildren = combineConsecutiveStrings(childNodes);
 
-  compactChildren.forEach((child, index) => {
+  for (let index = 0; index < compactChildren.length; index++) {
+    const child = compactChildren[index];
     if (isValidElement(child)) {
       currentGroup.push(child);
 
-      if (!isValidElement(compactChildren[index + 1])) {
+      const nextIndex = index + 1;
+      if (!isValidElement(compactChildren[nextIndex])) {
+        // If next child is the last string, add it to the current group
+        // to prevent it from being left hanging on the last line.
+        if (nextIndex === compactChildren.length - 1) {
+          const nextChild = compactChildren[nextIndex];
+          if (typeof nextChild === "string" && !!nextChild.trim()) {
+            currentGroup.push(
+              <Text key={`node-${nextIndex}`} ml="0.5rem">
+                {nextChild}
+              </Text>,
+            );
+          }
+          groupedNodes.push(
+            <div className={S.ControlGroup} key={`node-${index}`}>
+              {currentGroup}
+            </div>,
+          );
+          break;
+        }
```

Comment:
Note `isTextNodeLast` below. Would it work to tweak how that is used?

I think the string on the right needs to be centered vertically:

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/utGnP2ljFMMNOgi82vVn/b4f11148-ab48-46e4-b57c-250c4b367012.png)